src/org/sosy_lab/java_smt/example/ListAllSolversWithFeatures.java

```
@@ -0,0 +1,311 @@
+
+      info.add(solver.name());
+
+      info.add(context.getVersion());
+
+      info.add(getTheories(context));
+
+      info.add(getFeatures(context));
+    } catch (InvalidConfigurationException e) {
+      // Catches missing solvers
+    }
+
+    return info;
+  }
+
+  /**
+   * Checks the solver entered for features. Features checked: Optimization, UnsatCore,
+   * UnsatCoreWithAssumptions, Assumptions, AllSat, Interpolation
+   *
+   * @param context SolverContext you want to check for features.
+   * @return String with the features of the entered solver separated by a comma. Empty if none
+   *     available.
+   */
+  private String getFeatures(SolverContext context) throws SolverException, InterruptedException {
+    List<String> features = new ArrayList<>();
+
+    // Optimization: Will throw UnsupportedOperationException in creation of prover if not
+    // available.
+    try (OptimizationProverEnvironment prover =
+        context.newOptimizationProverEnvironment(ProverOptions.GENERATE_MODELS)) {
+      if (prover != null) {
+        features.add("Optimization");
+      }
+    } catch (UnsupportedOperationException e) {
+    }
+
+    // Interpolation: Will throw UnsupportedOperationException in creation of prover if not
+    // available.
+    try (InterpolatingProverEnvironment<?> prover =
+        context.newProverEnvironmentWithInterpolation(ProverOptions.GENERATE_MODELS)) {
+      if (prover != null) {
+        features.add("Interpolation");
+      }
+    } catch (UnsupportedOperationException e) {
+    }
+
+    // UnsatCore: throws UnsupportedOperationException if not available.
+    // TODO: check behavior with null argument properly
+    try (ProverEnvironment prover =
+        context.newProverEnvironment(ProverOptions.GENERATE_UNSAT_CORE)) {
+      if (prover.getUnsatCore() != null) {
+        features.add("UnsatCore");
+      }
+    } catch (UnsupportedOperationException e) {
```

add a comment, why the exception can be ignored.